sunpy/spectra/spectrogram.py

```diff
@@ -1260,4 +1260,8 @@ def in_interval(self, start=None, end=None):
                     *list(map(int, end.split(":")))
                 )
             end = self.time_to_x(end)
+        if start:
+            start = int(start)
+        if end:
+            end = int(end)
```

Comment:
These need to be moved into lines 1249 and 1262. The whole if statement is not needed.